### PR TITLE
description field and get by class

### PR DESCRIPTION
**Pull request by michael-grace**

### CODE REVIEW

File: subtype.go [hunk 1]
```diff
@@ -1,12 +1,30 @@
...
+	if err != nil {
+		return
+	}
+
+	for _, subtype := range subtypes {
+		if subtype.Class == class {
+			return subtype, err
+		}
+	}
+	return
```
Comment (markspolakovs) [inline subtype.go:29]: Worth clarifying in the doc comment that it can return nil?

**Comment by markspolakovs** [comment]
Oh, hang on, no, it can't return nil. It returns `ShowSubtype`, not `*ShowSubtype`, so it'll return the "zero value", which is that struct with all fields defaults (empty strings)

**Comment by michael-grace** [comment]
damn yeah, yet me see what i want

**Comment by michael-grace** [comment]
let's get it to return nil

**Comment by michael-grace** [comment]
ok, lets not get it to return nil

File: subtype.go [hunk 2]
```diff
@@ -16,16 +16,16 @@ func (s *Session) GetAllShowSubtypes() (subtypes []ShowSeasonSubtype, err error)
 
 // GetShowSubtypeByClass returns a ShowSeasonSubtype based on a given class
 // Can return nil if no subtype found for given class
-func (s *Session) GetShowSubtypeByClass(class string) (subtype ShowSeasonSubtype, err error) {
+func (s *Session) GetShowSubtypeByClass(class string) (ShowSeasonSubtype, error) {
```
Comment (markspolakovs) [inline subtype.go]:
```suggestion
func (s *Session) GetShowSubtypeByClass(class string) (*ShowSeasonSubtype, error) {
```

that'll let you return nil

File: subtype.go [hunk 3]
```diff
@@ -16,16 +16,16 @@ func (s *Session) GetAllShowSubtypes() (subtypes []ShowSeasonSubtype, err error)
...
 	subtypes, err := s.GetAllShowSubtypes()
 	if err != nil {
-		return
+		return nil, err
 	}
 
 	for _, subtype := range subtypes {
 		if subtype.Class == class {
 			return subtype, err
```
Comment (markspolakovs) [inline subtype.go:27]:
```suggestion
			return &subtype, err
```

along with that

**Comment by michael-grace** [comment]
nil pointer will be the best way, then can just check if it returns nil

**Pull request merged**